File: composeApp/src/commonMain/kotlin/org/hhp227/concafe/presentation/main/MainUiState.kt
package org.hhp227.concafe.presentation.main

import org.hhp227.concafe.domain.model.MainNavigationTab
import org.hhp227.concafe.domain.model.User

data class MainUiState(
    val currentUser: User? = null,
    val tabs: List<MainNavigationTab> = listOf(
        MainNavigationTab.HOME,
        MainNavigationTab.EXPLORE,
        MainNavigationTab.CHECK_IN,
        MainNavigationTab.RANKING,
        MainNavigationTab.MY_INFO
    ),
    val selectedTab: String = MainNavigationTab.HOME.route,
    val thirdTab: MainNavigationTab = MainNavigationTab.CHECK_IN
) {
    companion object {
        fun empty(): MainUiState = MainUiState()
    }
}

File: iosApp/iosApp/Presentation/Main/MainUiState.swift
//
//  MainUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/07.
//

import Foundation
import Shared

struct MainUiState {
    var currentUser: User?
    
    var tabs: [MainNavigationTab] = [
        .home,
        .explore,
        .checkIn,
        .ranking,
        .myInfo
    ]

    var selectedTab: String = MainNavigationTab.home.route

    var thirdTab: MainNavigationTab = .checkIn

    static let empty = MainUiState()
}
